tests/foreman/cli/test_host.py

```diff
@@ -79,3 +82,205 @@ def test_create_libvirt_without_mac(self):
+        })
+        self.host = Host.info({'id': self.host['id']})
+        self.assertIn(name, self.host['parameters'].keys())
+        self.assertEqual(value, self.host['parameters'][name])
+
+    def test_update_parameter_host_id(self):
+        """@Test: Update existing host parameter by specifying host ID.
+
+        @Feature: Hosts
+
+        @Assert: Host parameter was successfully updated with new value.
+
+        """
+        name = gen_string('alphanumeric').lower()
+        old_value = gen_string('alphanumeric')
+        Host.set_parameter({
+            'host-id': self.host['id'],
+            'name': name,
+            'value': old_value,
+        })
+        for new_value in valid_data_list():
+            with self.subTest(new_value):
+                Host.set_parameter({
+                    'host-id': self.host['id'],
+                    'name': name,
+                    'value': new_value,
+                })
+                self.host = Host.info({'id': self.host['id']})
+                self.assertIn(name, self.host['parameters'].keys())
+                self.assertEqual(new_value, self.host['parameters'][name])
+                self.assertNotEqual(old_value, self.host['parameters'][name])
+
+    def test_update_parameter_host_name(self):
+        """@Test: Update existing host parameter by specifying host name.
+
+        @Feature: Hosts
+
+        @Assert: Host parameter was successfully updated with new value.
+
+        """
+        name = gen_string('alphanumeric').lower()
+        old_value = gen_string('alphanumeric')
+        Host.set_parameter({
+            'host': self.host['name'],
+            'name': name,
+            'value': old_value,
+        })
+        for new_value in valid_data_list():
+            with self.subTest(new_value):
+                Host.set_parameter({
+                    'host': self.host['name'],
+                    'name': name,
+                    'value': new_value,
+                })
+                self.host = Host.info({'id': self.host['id']})
+                self.assertIn(name, self.host['parameters'].keys())
+                self.assertEqual(new_value, self.host['parameters'][name])
+                self.assertNotEqual(old_value, self.host['parameters'][name])
+
+    def test_delete_parameter_host_id(self):
+        """@Test: Delete existing host parameter by specifying host ID.
+
+        @Feature: Hosts
+
+        @Assert: Host parameter was successfully deleted.
+
+        """
+        name = gen_string('alphanumeric').lower()
+        Host.set_parameter({
+            'host-id': self.host['id'],
+            'name': name,
+            'value': gen_string('alphanumeric'),
+        })
+        Host.delete_parameter({
+            'host-id': self.host['id'],
+            'name': name,
+        })
+        self.host = Host.info({'id': self.host['id']})
+        self.assertNotIn(name, self.host['parameters'].keys())
+
+    def test_delete_parameter_host_name(self):
```

Comment:
no need to check it for `...delete_by_id` test, but for `...delete_by_name` it's good suggestion, will update the test.